src/Core/Resolvers/BaseQueryStructure.cs

```diff
@@ -181,5 +185,42 @@ internal static IObjectField ExtractItemsSchemaField(IObjectField connectionSche
         {
             return GraphQLUtils.UnderlyingGraphQLEntityType(connectionSchemaField.Type).Fields[QueryBuilder.PAGINATION_FIELD_NAME];
         }
+
+        internal static int GetPaginationLimit(RuntimeConfigProvider runtimeConfigProvider, int? userInputValue)
```

Comment:
It might be bloating the `RuntimeConfig` but helper methods which deal with all the intricacies and return the maxpagesize/defaultpagesize can be placed in the `RuntimeConfig`.

Reply:
This makes sense. moved it.